helpers/focus.js

```diff
@@ -62,6 +63,22 @@ export function getFocusableDescendants(node, options) {
 export function getFocusPseudoClass() {
 	return isFocusVisibleSupported() ? 'focus-visible' : 'focus';
 }
+export function getFocusRingStyles(selector, { applyOnHover = false, noFocusPseudoClass = false, baseOffset = '2px', extraStyles = '' } = {}) {
+	const baseSelector = selector;
+	if (!noFocusPseudoClass) selector += `:${getFocusPseudoClass()}`;
+	if (applyOnHover) selector += `, ${baseSelector}:hover`;
```

Comment:
To better handle these complex selector cases where `noFocusPseudoClass` is set, instead of a `selector` we could maybe instead have consumers pass in a delegate that gets called with the pseudo-class?

Simple example:
```javascript
getFocusRingStyles(
  pseudoClass => `#d2l-tooltip-help-text:${pseudoClass}`, {
    baseOffset: '0.05rem',
    borderRadius: '0.05rem'
  }
);
```

Complex case:
```javascript
getFocusRingStyles(
  pseudoClass => `:host(:${pseudoClass}) .d2l-tab-text-inner-content`, {
    ...
  }
);
```